File: 12. Rekursi/B. Faktorial Ganjil-Genap/b_faktorial_ganjil_genap.go
package main

import (
	"fmt"
)

func factorial_gg(N int) int{
	if N == 1{
		return 1
	} else if N % 2 == 0{
		return N / 2 * factorial_gg(N-1)
	} else{
		return N * factorial_gg(N-1)
	}
}

func main() {
	var N int
	fmt.Scan(&N)
	fmt.Println(factorial_gg(N))
}

File: 12. Rekursi/B. Faktorial Ganjil-Genap/b_faktorial_ganjil_genap.java
import java.util.Scanner;

public class b_faktorial_ganjil_genap {
    public static Integer factorial_gg(Integer N) {
        if (N == 0) {
            return 1;
        } else if (N % 2 == 0) {
            return N / 2 * factorial_gg(N - 1);
        } else {
            return N * factorial_gg(N - 1);
        }
    }

    public static void main(String args[]) {
        Scanner scan = new Scanner(System.in);
        Integer N = scan.nextInt();
        scan.close();
        System.out.println(factorial_gg(N));
    }
}
